### DISCUSSION TITLE
Create SBOM using github action in workflow

### DISCUSSION THREAD

**Discussion by perifitdevs** [codemagic-ci-cd/codemagic-docs#2035] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

React Native

### Steps to reproduce

Hello,
I've been trying to create SBOM (Software Bill of Materials) using github action anchore/sbom-action@v0 (https://github.com/anchore/sbom-action) in workflow:

```
workflows:
    build-sbom:
        name: Create SBOM
        uses: anchore/sbom-action@v0
        with:
            path: ./
            format: spdx
            artifact-name: myapp.spdx
```

But it seems that's not enough to make it work, I got below error when running it from Codemagic:
_invalid configuration: 2 validation errors for YamlConfiguration workflows -> build-sbom -> uses extra fields not permitted (type=value_error.extra) workflows -> build-sbom -> with extra fields not permitted (type=value_error.extra)_

I'm a newbie with Codemagic.
Can anybody help on this ?

Thanks and Best Regards

### Expected results

I expect it can run the github action without error.

### Actual results

I got below error when running it from Codemagic:
_invalid configuration: 2 validation errors for YamlConfiguration workflows -> build-sbom -> uses extra fields not permitted (type=value_error.extra) workflows -> build-sbom -> with extra fields not permitted (type=value_error.extra)_

**Comment by himesh-cm** [comment, marked as answer]
Hi @perifitdevs, Please note that the GitHub Actions script won't work as is in `codemagic.yaml` file. Please check the documentation [here](https://docs.codemagic.io/yaml-basic-configuration/yaml-getting-started/) on using `codemagic.yaml` configuration for your workflows.

Regarding the implementation of SBOM, you can directly follow the instruction process mentioned on their GitHub repo [here](https://github.com/anchore/syft) by defining the scripts in the `codemagic.yaml` file, you can find a few examples [here](https://docs.codemagic.io/integrations/jira-integration/) for reference.

**Reply by perifitdevs** [reply]
Thanks a lot @himesh-cm !

**Reply by himesh-cm** [reply]
Happy to help
While I’m closing our discussion for now, if you have any other questions or suggestions, feel free to reply back or start a new discussion anytime. Have a nice day!
